### DISCUSSION TITLE
Separate seurat object by samples

### DISCUSSION THREAD

**Discussion by devin-qiu** [satijalab/seurat#5234] [Q&A] [answered]
I have a set of matrix, features and barcodes files created by cellranger, where all samples are integrated together. I have read them into a seurat object and would like to call out different samples according to their sample ids. How should I do that? 

When I ran this: 

seurat_whole <- CreateSeuratObject(counts = whole, 
                             min.cells = 3,
                             project = "whole")

It looks like all the orig.ident became labeled as "whole". Is there a better way to assign id to different samples?

**Comment by cswoboda** [comment, marked as answer]
Depends on what you have prior!

So, if I'm reading this correctly, you have three independent count matrices that you merge into a "whole" count matrices prior to creating the seurat object seurat_whole.

An easy fix if this is the case is create a seurat object for each sample and then merge after.

A <-  CreateSeuratObject(counts = A_counts, min.cells = 3, project = "A")
B <-  CreateSeuratObject(counts = B_counts, min.cells = 3, project = "B")
C <-  CreateSeuratObject(counts = C_counts, min.cells = 3, project = "C")

whole <- merge(A, c(B, C))

If you know the order they're merged in, then you can simply add into the metadata like this:

seurat_whole$orig.ident <- c(rep("A", nA), rep("B", nB), rep("C", nC))

where nA, nB, nC are equal to the number of cells from each dataset included in the seurat_whole. Important note, these HAVE to be in the order they're merged together in the counts. So if you added B to C to A lets say, you'd need to do: 
seurat_whole$orig.ident <- c(rep("B", nB), rep("C", nC), rep("A", nA))

**Reply by devin-qiu** [reply]
Thank you so much!

**Reply by cswoboda** [reply]
Of course! Good luck with the rest of analysis :)

**Reply by KelvinHYYip** [reply]
Hi, I am new to scRNAseq analysis. I recently started analysing a 10X scRNA seq experiment using Seurat.

If I have multiple samples (3 different genotypes, 2 biological replicate each), combined all the samples and create a single Seurat object and followed by comparative analysis e.g. DEG between genotypes based on the meta data orig.ident, why would I need to use the IntegrateData function then?

I have tried create both a single Seurat object from all the samples and Seurat objects individually, the clusters look different on UMAP plots.

Can someone clarify this? Thank you.
